ggshield/utils/git_shell.py

```diff
@@ -214,7 +214,13 @@ def git(
         logger.debug("command=%s timeout=%d", command, timeout)
         result = subprocess.run(
             (
-                [_get_git_path(), "-c", "core.quotePath=false"]
+                [
+                    _get_git_path(),
+                    "-c",
+                    "core.quotePath=false",
+                    "-c",
+                    "safe.directory=*",
```

Comment:
Would it work to use `.` instead of `*` or does `safe.directory` have to be the root of the working tree? It would feel safer to use `.`.

Reply:
OK, let's keep that way then.

I just realized however that this changes makes it possible to simplify our docker config quite a bit: we don't need [docker/entrypoint.sh](https://github.com/GitGuardian/ggshield/blob/main/docker/entrypoint.sh) to create a custom config file with `safe.directory` in it anymore, which means I think we  don't need entrypoint.sh anymore at all!

Can you try removing it? Removing it as the additional advantage that the "dubious permission" error is checked for Docker scans (by tests/functional/test_docker_image.py), so this would ensure your changes are tested.